src/ui/pages/SwapForm/SlippageSettings/SlippageSettings.tsx

```diff
@@ -91,6 +95,67 @@ function SlippageWarning({ percentValue }: { percentValue: string }) {
   );
 }
 
+function PercentWidth({ onValue }: { onValue(width: number): void }) {
+  const persentCharWidthRef = useRef<HTMLDivElement | null>(null);
+
+  useEffect(() => {
+    let interval: ReturnType<typeof setTimeout>;
+    if (persentCharWidthRef.current) {
+      onValue(persentCharWidthRef.current.clientWidth);
+    } else {
+      interval = setTimeout(
+        () => onValue(persentCharWidthRef.current?.clientWidth || 0),
+        100
+      );
+    }
+    return () => clearTimeout(interval);
+  }, [onValue]);
+
+  return (
+    <UIText
+      ref={persentCharWidthRef}
+      kind="body/accent"
+      style={{ visibility: 'hidden', position: 'absolute' }}
+    >
+      %
+    </UIText>
+  );
+}
+
+function CustomValueOverlay({
+  isOptimal,
+  value,
+}: {
+  isOptimal: boolean;
+  value: string | null;
+}) {
+  return (
+    <div
+      style={{
+        position: 'absolute',
+        inset: '0 0 0 0',
+        pointerEvents: 'none',
+        display: 'flex',
+        justifyContent: 'center',
+        alignItems: 'center',
+      }}
+    >
+      {value ? (
+        <UIText
+          kind="body/accent"
+          color={isOptimal ? undefined : 'var(--notice-500)'}
+        >
+          {value}%
+        </UIText>
+      ) : (
+        <UIText kind="body/accent" color="var(--neutral-500)">
+          Custom
```

Comment:
Maybe hide placeholder when input is focused? Otherwise the cursor appears behind text: 
<img width="207" alt="image" src="https://github.com/zeriontech/zerion-wallet-extension/assets/5347023/8315bf27-2d67-40af-8043-6cffd03d90c1">